types/defaultTo.d.ts

```diff
@@ -1,2 +1,12 @@
-export function defaultTo<T, U>(a: T, b: U | null | undefined): T | U;
-export function defaultTo<T>(a: T): <U>(b: U | null | undefined) => T | U;
+
+export function defaultTo<Default, Value>(a: Default, b: Value): (Value extends (null | undefined) ? Default | Exclude<Value, null | undefined> : Value);
+export function defaultTo<Default>(a: Default): <Value>(b: Value) => (Value extends (null | undefined) ? Default | Exclude<Value, null | undefined> : Value);
+
+
+type DefaultTo<Default, Value> = (Value extends (null | undefined) ? Default | Exclude<Value, null | undefined> : Value);
```

Comment:
Yeah, good call.  I actually started to look at `propOr` yesterday and I think we might want to keep `DefaultTo` around to use in definitions like `propOr`.

Regarding that, I was wondering about splitting `tools.d.ts` into separate files, one for each helper.     
It would make the dependencies of each one more clear.  eg.  Any thoughts on that?